core/trino-main/src/main/java/io/trino/server/protocol/Query.java

```diff
@@ -575,6 +577,7 @@ private synchronized QueryResultRows removePagesFromExchange(QueryInfo queryInfo
             }
             if (exchangeDataSource.isFinished()) {
                 exchangeDataSource.close();
+                serde = null; // null to reclaim memory
```

Comment:
I think ideally it would be great to account for PageSerde memory in operators. A single `PageSerde` can retain up to 128KB of memory (when both encryption and compression are enabled). Hard to say how many open Exchange/TaskOutput operators can be open at the same time, but in theory it can accumulate. 

Created an issue: https://github.com/trinodb/trino/issues/15273

Reply:
I made comment more descriptive